src/coreclr/tools/superpmi/superpmi/streamingsuperpmi.cpp

```diff
@@ -0,0 +1,244 @@
+            break;
+        }
+
+        LogDebug("[streaming] Method %d", index);
+
+        LightWeightMap<DWORD,DWORD>* baseForceJitOptions = o.forceJitOptions;
+        LightWeightMap<DWORD,DWORD>* forceJitOptions = nullptr;
+        bool skip = false;
+
+        while (tok = strtok_s(nullptr, seps, &next))
+        {
+            if (forceJitOptions == nullptr)
+            {
+                if (baseForceJitOptions == nullptr)
+                {
+                    forceJitOptions = new LightWeightMap<DWORD, DWORD>();
+                }
+                else
+                {
+                    forceJitOptions = new LightWeightMap<DWORD, DWORD>(*baseForceJitOptions);
+                }
+            }
+
+            bool added = AddJitOption(forceJitOptions, tok);
+
+            if (!added)
+            {
+                LogInfo("[streaming] unable to parse option '%s'", tok);
+                skip = true;
+                break;
+            }
+        }
+
+        if (skip)
+        {
+            continue;
+        }
+
+        LogDebug("[streaming] Launching...");
+
+        // launch this instance...
+        //
+        reader->Reset(&index, 1);
+        MethodContextBuffer mcb = reader->GetNextMethodContext();
+
+        if (mcb.Error())
+        {
+            return (int)SpmiResult::GeneralFailure;
+        }
+
+        MethodContext* mc = nullptr;
+        if (!MethodContext::Initialize(index, mcb.buff, mcb.size, &mc))
+        {
+            return (int)SpmiResult::GeneralFailure;
+        }
+
+        if (jit == nullptr)
+        {    
+            SimpleTimer stInitJit;
+            jit = JitInstance::InitJit(o.nameOfJit, o.breakOnAssert, &stInitJit, mc, forceJitOptions, o.jitOptions);
+            
+            if (jit == nullptr)
+            {
+                // InitJit already printed a failure message
+                return (int)SpmiResult::JitFailedToInit;
+            }
+        }
+        else
+        {
+            jit->updateForceOptions(forceJitOptions);
+            jit->resetConfig(mc);
+        }
+
+        bool collectThroughput = false;
+        ReplayResults res = jit->CompileMethod(mc, reader->GetMethodContextIndex(), collectThroughput);
+
+        // Protocol with clients is for them to read stdout. Let them know we're done.
+        //
+        printf("[streaming] Done.\n");
+        fflush(stdout);
+
+        if (res.Result == ReplayResult::Success)
+        {
+            if (Logger::IsLogLevelEnabled(LOGLEVEL_DEBUG))
+            {
+                mc->cr->dumpToConsole(); // Dump the compile results if doing debug logging
+            }
+        }
+        else if (res.Result == ReplayResult::Error)
+        {
+            LogError("Method %d of size %d failed to load and compile correctly%s (%s).",
+                reader->GetMethodContextIndex(), mc->methodSize,
+                (o.nameOfJit2 == nullptr) ? "" : " by JIT1", o.nameOfJit);
+            return (int)SpmiResult::Error;
```

Comment:
Should it return on first failure? Shouldn't it keep going until streaming is done, and return overall failure if any single replay returned failure?

Reply:
I'm not sure a return value from the server process is particularly useful, but it would be nice to have some sort of error signal sent back to the client. Maybe a status code on the done message.

Reply:
Added a code